tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/ServerlessMiniAgent.cs

```diff
@@ -0,0 +1,164 @@
+    {
+        if (!settings.IsRunningInGCPFunctions && !settings.IsRunningInAzureFunctionsConsumptionPlan)
+        {
+            return null;
+        }
+
+        string rustBinaryPath;
+        if (Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH") != null)
+        {
+            rustBinaryPath = Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH");
+        }
+        else
+        {
+            // Environment.OSVersion.Platform can return PlatformID.Unix on MacOS, this is OK as GCP & Azure don't have MacOs functions.
+            if (Environment.OSVersion.Platform != PlatformID.Unix && Environment.OSVersion.Platform != PlatformID.Win32NT)
+            {
+                Log.Error("Serverless Mini Agent is only supported on Windows and Linux.");
+                return null;
+            }
+
+            var isWindows = os == PlatformID.Win32NT;
+            var dirPathSep = isWindows ? "\\" : "/";
+
+            string rustBinaryPathRoot;
+            if (settings.IsRunningInGCPFunctions)
+            {
+                rustBinaryPathRoot = "/layers/google.dotnet.publish/publish/bin";
+            }
+            else
+            {
+                // IsAzureFunction
+                if (isWindows)
+                {
+                    rustBinaryPathRoot = "C:\\home\\site\\wwwroot";
+                }
+                else
+                {
+                    // linux
+                    rustBinaryPathRoot = "/home/site/wwwroot";
+                }
+            }
+
+            string rustBinaryPathOsFolder = isWindows ? "datadog-serverless-agent-windows-amd64" : "datadog-serverless-agent-linux-amd64";
+            string rustBinaryFileExtension = isWindows ? ".exe" : string.Empty;
+            rustBinaryPath = string.Format("{0}{1}{2}{3}datadog-serverless-trace-mini-agent{4}", rustBinaryPathRoot, dirPathSep, rustBinaryPathOsFolder, dirPathSep, rustBinaryFileExtension);
+        }
+
+        return rustBinaryPath;
+    }
+
+    internal static void StartServerlessMiniAgent(ImmutableTracerSettings settings)
+    {
+        var serverlessMiniAgentPath = ServerlessMiniAgent.GetMiniAgentPath(Environment.OSVersion.Platform, settings);
+        if (string.IsNullOrEmpty(serverlessMiniAgentPath))
+        {
+            return;
+        }
+
+        if (!File.Exists(serverlessMiniAgentPath))
+        {
+            Log.Error("Serverless Mini Agent does not exist: {Path}", serverlessMiniAgentPath);
+            return;
+        }
+
+        try
+        {
+            Log.Debug("Trying to spawn the Serverless Mini Agent at path: {Path}", serverlessMiniAgentPath);
+            Process process = new Process();
+            process.StartInfo.FileName = serverlessMiniAgentPath;
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.OutputDataReceived += new DataReceivedEventHandler(MiniAgentDataReceivedHandler);
```

Comment:
[coding style] Using the delegate constructor syntax is redundant and rarely used.
```suggestion
            process.OutputDataReceived += MiniAgentDataReceivedHandler;
```